### PR TITLE
coprocessor: skip transient read request

### LINKED ISSUE #15405: coprocessor: don't report the transient flow to pd
## Feature Request
In past, coprocess will collect all request read flow and report to pd, but some request is transient flow, pd can't balance this flow, and it brings many unstable problems like this:
img_v2_6a2f0f8d-8779-4063-bbdd-7a29925f0cag

### Is your feature request related to a problem? Please describe:

hot region scheduler can‘t work well.
### Describe alternatives you've considered:

ignore the transient read requestion

### PR DESCRIPTION
**Pull request by bufferflies**
### What is changed and how it works?

Issue Number: Close #15405

What's Changed:

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

image

image

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by SpadeA-Tang — APPROVED** [review]

File: src/coprocessor/tracker.rs [hunk 1]
```diff
@@ -350,20 +350,24 @@ impl<E: Engine> Tracker<E> {
...
+        // like analyze and checksum.
+        if self.req_ctx.tag == ReqTag::select || self.req_ctx.tag == ReqTag::index {
+            tls_collect_query(
+                region_id,
+                peer,
+                start_key.as_encoded(),
+                end_key.as_encoded(),
+                reverse_scan,
+            );
+            tls_collect_read_flow(
```
Comment (nolouch) [inline src/coprocessor/tracker.rs:363]: I wonder if it affects the coprocess metrics in the TiKV Panel?

Comment (bufferflies) [reply src/coprocessor/tracker.rs:363]: no, coprocess metrics doesn't depends this, it depends in coprocess module.

File: src/coprocessor/tracker.rs [hunk 2]
```diff
@@ -350,20 +350,24 @@ impl<E: Engine> Tracker<E> {
...
-        tls_collect_read_flow(
-            self.req_ctx.context.get_region_id(),
-            Some(start_key.as_encoded()),
-            Some(end_key.as_encoded()),
-            &total_storage_stats,
-            self.buckets.as_ref(),
-        );
+        // only collect metrics for select and index, exclude transient read flow such
+        // like analyze and checksum.
+        if self.req_ctx.tag == ReqTag::select || self.req_ctx.tag == ReqTag::index {
```
Comment (lhy1024) [inline src/coprocessor/tracker.rs:355]: Are store and region statistics updated here?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM, but ci failed

File: src/coprocessor/tracker.rs (hunk 2 above)
Comment (bufferflies) [reply src/coprocessor/tracker.rs:355]: yes

**Pull request merged**